core/src/avm1.rs

```diff
@@ -173,6 +173,12 @@ impl<'gc> Avm1<'gc> {
     /// 
     /// Stack frame identity (for the purpose of the above paragraph) is
     /// determined by the data pointed to by the `SwfSlice` of a given frame.
+    /// 
+    /// # Warnings
+    /// 
+    /// It is incorrect to call this function multiple times in the same stack.
```

Comment:
Shouldn't this be enforced, then? At least in development builds.

Reply:
I'm not sure what the appropriate way to do this. The best I could think of would be to set a flag in the AVM and use that to panic, but I'm not sure if that's a good idea.

Reply:
You can detect dev / test builds with `if cfg!` and have the flag only in those… but I don't know if that's good practice.

Alternatively, you could have a helper function / macro with a dummy implementation when `!cfg(debug_assertions)`, which seems safer.

Reply:
I implemented something that just logs an error message if this particular situation occurs for now.